lib/IRGen/EnumPayload.cpp

```diff
@@ -564,6 +564,9 @@ EnumPayload::emitCompare(IRGenFunction &IGF, APInt mask, APInt value) const {
       v = IGF.Builder.CreateBitOrPointerCast(v, intTy);
     
     if (isMasked) {
+      if( !IGF.IGM.Triple.isLittleEndian() && IGF.IGM.Triple.isArch64Bit() ){
+        v  = IGF.Builder.CreateLShr(v, 56);
+      }
```

Comment:
This will only work for 64-bit platforms.

Reply:
This should probably be a byte-swap rather than a shift. There may be significant bits in the low 56 bits of the mask piece..

Reply:
@jckarter as mentioned in https://github.com/apple/swift/pull/19832#issuecomment-440318674, will work on this.